k8s-canary-rollout.sh

```diff
@@ -29,31 +28,37 @@ healthcheck(){
 
 cancel(){
     echo "[CANARY] Cancelling rollout - healthcheck failed"
-    
+
     echo "[CANARY SCALE] Restoring original deployment to $PROD_DEPLOYMENT"
     kubectl apply -f $WORKING_VOLUME/original_deployment.yaml -n $NAMESPACE
     kubectl rollout status deployment/$PROD_DEPLOYMENT
 
     #we could also just scale to 0.
-    echo "[CANARY DELETE] Removing canary completely"
+    echo "[CANARY DELETE] Removing canary deployment completely"
     kubectl delete deployment $CANARY_DEPLOYMENT
 
+    echo "[CANARY DELETE] Removing canary horizontal pod autoscaler completely"
+    kubectl delete deployment $CANARY_DEPLOYMENT -n $NAMESPACE
+
     exit 1
 }
 
-
 cleanup(){
     echo "[CANARY CLEANUP] removing previous deployment $PROD_DEPLOYMENT"
     kubectl delete deployment $PROD_DEPLOYMENT -n $NAMESPACE
-    echo "[CANARY CLEANUP] marking canary as new production"
-    kubectl get service $SERVICE_NAME -o=yaml --namespace=${NAMESPACE} | sed -e "s/$CURRENT_VERSION/$NEW_VERSION/g" | kubectl apply --namespace=${NAMESPACE} -f - 
 
+    echo "[CANARY CLEANUP] removing previous horizontal pod autoscaler $PROD_DEPLOYMENT"
+    kubectl delete hpa $PROD_DEPLOYMENT -n $NAMESPACE
```

Comment:
Will this command succeed if the user of the canary plugin does NOT have HPA running? Maybe we need an global variable to decided if the HPA was present or not before the rolling update?

Reply:
the command  will fail, but this bash  allows for exit codes > 0. There isn't  much error checking in the script in general. (like you can define a DEPLOYMENT that doesn't exist and the  script will chug along. so I'm just sticking to the current conventions)